### PR TITLE
Sync merge GFSv16.2.1 updates into dev_v16

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
**Description**

This is a sync merge PR of GFSv16.2.1 updates from `feature/ops-wcoss2` into its child branch `dev_v16`. See PRs #874 and #879 for more details on incoming vetted/tested changes.

**Type of change**

Sync merge

**How Has This Been Tested?**

Tested and vetted by NCO, GDIT, and EMC in corresponding PRs mentioned above.

Refs #665

### CODE REVIEW

File: sorc/gfs_bufr.fd/calpreciptype.f [hunk 1]
```diff
@@ -77,7 +77,7 @@ SUBROUTINE CALPRECIPTYPE(kdt,nrcm,im,ix,lm,lp1,randomno,      &
       ALLOCATE ( RH(LM),TD8(LM),TWET8(LM) )
 
 ! Create look up table
-      call gfuncphys
+!     call gfuncphys
```
Comment (WalterKolczynski-NOAA) [inline sorc/gfs_bufr.fd/calpreciptype.f:80]: Remove inactive code?

File: sorc/gfs_bufr.fd/meteorg.f [hunk 2]
```diff
@@ -63,6 +63,7 @@ subroutine meteorg(npoint,rlat,rlon,istat,cstat,elevstn,
       use sigio_module 
       use physcons
       use mersenne_twister
+!      use funcphys, only : gfuncphys
```
Comment (WalterKolczynski-NOAA) [inline sorc/gfs_bufr.fd/meteorg.f:66]: Remove inactive code?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: sorc/gfs_bufr.fd/calpreciptype.f (hunk 1 above)
Comment (KateFriedman-NOAA) [reply sorc/gfs_bufr.fd/calpreciptype.f:80]: I can remove it once it's in `dev_v16` (for GFSv16.3+). I already handed off the v16.2.1 tag to NCO so I shouldn't touch `feature/ops-wcoss2` at this point without changes in ops.

File: sorc/gfs_bufr.fd/meteorg.f (hunk 2 above)
Comment (KateFriedman-NOAA) [reply sorc/gfs_bufr.fd/meteorg.f:66]: Same comment reply as above.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**